indexer/string_file_values.hpp

```diff
@@ -18,138 +23,236 @@
 
-  inline size_t size() const { return sizeof(m_value); }
+  inline size_t size() const { return sizeof(m_featureId); }
 
-  bool operator<(FeatureIndexValue const & value) const { return m_value < value.m_value; }
+  bool operator<(FeatureIndexValue const & o) const { return m_featureId < o.m_featureId; }
 
-  bool operator==(FeatureIndexValue const & value) const { return m_value == value.m_value; }
+  bool operator==(FeatureIndexValue const & o) const { return m_featureId == o.m_featureId; }
 
-  void swap(FeatureIndexValue & value) { ::swap(m_value, value.m_value); }
+  void Swap(FeatureIndexValue & o) { ::swap(m_featureId, o.m_featureId); }
 
-  uint32_t m_value;
+  uint64_t m_featureId;
 };
 
-/// A wrapper around serialized SaverT::ValueType.
-struct SerializedFeatureInfoValue
+struct FeatureWithRankAndCenter
 {
-  using ValueT = buffer_vector<uint8_t, 32>;
+  FeatureWithRankAndCenter() = default;
+
+  FeatureWithRankAndCenter(m2::PointD pt, uint32_t featureId, uint8_t rank,
+                           serial::CodingParams codingParams)
+    : m_pt(pt), m_featureId(featureId), m_rank(rank), m_codingParams(codingParams)
+  {
+  }
 
   template <typename TWriter>
-  void Write(TWriter & writer) const
+  void Serialize(TWriter & writer) const
   {
-    rw::WriteVectorOfPOD(writer, m_value);
+    serial::SavePoint(writer, m_pt, m_codingParams);
+    WriteToSink(writer, m_featureId);
+    WriteToSink(writer, m_rank);
   }
 
   template <typename TReader>
-  void Read(TReader & reader)
+  void Deserialize(TReader & reader)
   {
-    rw::ReadVectorOfPOD(reader, m_value);
+    ReaderSource<TReader> src(reader);
+    DeserializeFromSource(src);
   }
 
-  inline void const * data() const { return m_value.data(); }
+  template <typename TSource>
+  void DeserializeFromSource(TSource & src)
+  {
+    m_pt = serial::LoadPoint(src, m_codingParams);
+    m_featureId = ReadPrimitiveFromSource<uint32_t>(src);
+    m_rank = ReadPrimitiveFromSource<uint8_t>(src);
+  }
 
-  inline size_t size() const { return m_value.size() * sizeof(ValueT::value_type); }
+  bool operator<(FeatureWithRankAndCenter const & o) const { return m_featureId < o.m_featureId; }
 
-  bool operator<(SerializedFeatureInfoValue const & value) const { return m_value < value.m_value; }
+  bool operator==(FeatureWithRankAndCenter const & o) const { return m_featureId == o.m_featureId; }
 
-  bool operator==(SerializedFeatureInfoValue const & value) const
+  void Swap(FeatureWithRankAndCenter & o)
   {
-    return m_value == value.m_value;
+    swap(m_pt, o.m_pt);
+    swap(m_featureId, o.m_featureId);
+    swap(m_rank, o.m_rank);
   }
 
-  void swap(SerializedFeatureInfoValue & value) { m_value.swap(value.m_value); }
+  void SetCodingParams(serial::CodingParams const & codingParams) { m_codingParams = codingParams; }
 
-  ValueT m_value;
+  m2::PointD m_pt;       // Center point of the feature.
+  uint32_t m_featureId;  // Offset of the feature.
```

Comment:
It's not "offset" - it can be and index of the feature, for example. So it's better to use "Feature identifier" instead of "offset".